File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.py
def solution(arr):
    answer = arr
    n = 0
    # 2의 배수보다 작을 때
    while True:
        if len(arr) <= 2**n:
            break
        n += 1
    
    for i in range(2**n-len(arr)):
        answer.append(0)
    
    return answer

File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.java
class Solution {
    public int[] solution(int[] arr) {
        int cnt = 1;
        int len = arr.length;
        
        while (cnt < len) {
            if (cnt < len) {
                cnt *= 2;
            }
        }
        
        int[] answer = new int[cnt];
        
        for (int i = 0; i < cnt; i++) {
            if (i < len) {
                answer[i] = arr[i];
            } else {
                answer[i] = 0;
            }
        }
        
        
        
        return answer;
    }
}